### ISSUE TITLE
[Bug]: Syncing error

### ISSUE THREAD

**Issue by YIRU69** [sboesen/remotely-secure#23]
### What happened?

1. delete a file `tracked_files.json` . close the plugin
2. generate `tracked_files.json` under the other folder
3. manual sync
4. syncing error.

```
Error: do not know which level for //tracked_files.json
    at HE (plugin:remotely-secure:49:26549)
    at u5e (plugin:remotely-secure:112:49395)
    at Qre (plugin:remotely-secure:112:50341)
    at async Rx.syncRun (plugin:remotely-secure:180:3091)
syncRun @ plugin:remotely-secure:180

```

### What OS are you using?

Windows

### What remote cloud services are you using?

S3

### Version of the plugin

5.15

### Version of Obsidian

1.4.16

### Using password or not

- [X] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by YIRU69** [comment]
```
    "tracked_files.json": {
      "key": "tracked_files.json",
      "existRemote": true,
      "mtimeRemote": 1700185121000,
      "mtimeRemoteFmt": "2023-11-17T09:38:41+08:00",
      "sizeRemoteEnc": 670917,
      "remoteEncryptedKey": "NoZ1fTdrlaJdDGXvWRpjXpZwgDI3Gi47bL0mTfyyku_3cpvXlwO8z5B4VI57DudWtAtnZKIqtpRs_3l_Wrw",
      "changeRemoteMtimeUsingMapping": false,
      "existLocal": false,
      "deltimeRemote": 1700215125304,
      "deltimeRemoteFmt": "2023-11-17T17:58:45+08:00",
      "deltimeLocal": 1700215125304,
      "deltimeLocalFmt": "2023-11-17T17:58:45+08:00",
      "decision": "uploadLocalDelHistToRemote",
      "decisionBranch": 39
    },
    "700-Extras/tracked_files.json": {
      "key": "700-Extras/tracked_files.json",
      "deltimeRemote": 1700230853671,
      "deltimeRemoteFmt": "2023-11-17T22:20:53+08:00",
      "existLocal": false,
      "existRemote": false,
      "deltimeLocal": 1700230853671,
      "deltimeLocalFmt": "2023-11-17T22:20:53+08:00",
      "decision": "uploadLocalDelHistToRemote",
      "decisionBranch": 39
    },
```

**Comment by sboesen** [comment]
Is this `tracked_files` just some random file as an example?

So I understand, steps to reproduce:

1. Create a file with name A
2. Delete the file (inside obsidian? or outside?)
3. Create a file with the same name A in another folder
4. Sync

Or am I misunderstanding?

**Comment by YIRU69** [comment]
Yeah, but, I don't know this step can 100% reproduce.

**Comment by YIRU69** [comment]
Now, I can't sync because the error. Can you add a forced upload feature to fix this? Here's a workaround I'd like to have, if not reproduced

**Comment by sboesen** [comment]
Updated 0.4.16 so it doesn't block sync and instead logs the error. Please let me know if this is still not letting you sync, sorry about that :( 

I'll give the reproduction steps a try and attempt to improve the sync logic here - tracked in new issue #25.

**Comment by YIRU69** [comment]
It also block sync.

**Comment by sboesen** [comment]
Another fix attempted in 8ebd765 (0.4.17). Not sure why that is blocking sync, but fixed another thing that would probably have prevented it from syncing as well (not skipping the erroring file).

**Comment by YIRU69** [comment]
It worked!

**Comment by sboesen** [comment]
Awesome! Will close this issue for now, but it doesn't seem like the root issue is addressed, so definitely keeping #25 open for now.
